taichi_three/geometry.py

```diff
@@ -11,48 +11,48 @@ def render_triangle(model, face):
     ia, ib, ic = model.vi[face[0, 0]], model.vi[face[1, 0]], model.vi[face[2, 0]]
     ta, tb, tc = model.vt[face[0, _1]], model.vt[face[1, _1]], model.vt[face[2, _1]]
     na, nb, nc = model.vn[face[0, _2]], model.vn[face[1, _2]], model.vn[face[2, _2]]
-    a = scene.camera.untrans_pos(L2W @ ia)
-    b = scene.camera.untrans_pos(L2W @ ib)
-    c = scene.camera.untrans_pos(L2W @ ic)
-    A = scene.uncook_coor(a)
-    B = scene.uncook_coor(b)
-    C = scene.uncook_coor(c)
-    B_A = B - A
-    C_B = C - B
-    A_C = A - C
-    ilB_A = 1 / ts.length(B_A)
-    ilC_B = 1 / ts.length(C_B)
-    ilA_C = 1 / ts.length(A_C)
-    B_A *= ilB_A
-    C_B *= ilC_B
-    A_C *= ilA_C
-    BxA = ts.cross(B, A) * ilB_A
-    CxB = ts.cross(C, B) * ilC_B
-    AxC = ts.cross(A, C) * ilA_C
-    normal = ts.normalize(ts.cross(a - c, a - b))
-    light_dir = scene.camera.untrans_dir(scene.light_dir[None])
-    center_pos = (a + b + c) / 3
+    for camera in ti.static(scene.cameras):
```

Comment:
Can we use
`def render_triangle(camera, model, face):`
and move this for-loop to the caller?

Reply:
Yes, I will move it.